src/aa_remove_data/remove_data.py

```diff
@@ -0,0 +1,486 @@
+import argparse
+import time
+from collections.abc import Callable
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+from aa_remove_data.pb_utils import DEFAULT_CHUNK_SIZE, PBUtils
+
+
+def get_nano_diff(sample1: type, sample2: type) -> int:
+    """Get the difference in nano seconds between two samples.
+
+    Args:
+        sample1 (type): An Archiver Appliance sample.
+        sample2 (type): Another Archiver Appliance sample.
+
+    Returns:
+        int: Difference in nanoseconds.
+    """
+    diff = (sample2.secondsintoyear - sample1.secondsintoyear) * 10**9 + (
+        sample2.nano - sample1.nano
+    )
+    assert diff > 0
+    return diff
+
+
+def get_seconds_diff(sample1: type, sample2: type) -> int:
+    """Get the difference in whole seconds between two samples.
+
+    Args:
+        sample1 (type): An Archiver Appliance sample.
+        sample2 (type): Another Archiver Appliance sample.
+
+    Returns:
+        _type_: Difference in seconds
+    """
+    diff = sample2.secondsintoyear - sample1.secondsintoyear
+    assert diff >= 0
+    return diff
+
+
+def benchmark(function: Callable, *args: Any, **kwargs: Any) -> Any:
+    """Print the time taken for a function to execute.
+
+    Args:
+        function (Callable): Function to execute.
+
+    Returns:
+        Any: Result of the function.
+    """
+    start_time = time.time()
+    result = function(*args, **kwargs)
+    end_time = time.time()
+    print(f"Time taken for {function.__name__}: {end_time - start_time:.2f}s.")
+    return result
+
+
+def reduce_freq(
+    samples: list,
+    freq: float = 0,
+    period: float = 0,
+    initial_sample: type | None = None,
+) -> list:
+    """Reduce the frequency of a list of samples. Specify the desired frequency
+    or period (not both).
+
+    Args:
+        samples (list): List of samples
+        freq (float, optional): Desired frequency. Defaults to 0.
+        period (float, optional): Desired period. Defaults to 0.
+        initial_sample (type, optional): An initial sample to find an initial diff.
+
+    Returns:
+        list: Reduced list of samples
+    """
+    assert freq * period == 0 and (freq + period) > 0, (
+        "Must set either frequency or period, not both or none."
+    )
+    if freq:
+        seconds_delta = 1 / freq
+    else:
+        seconds_delta = period
+    nano_delta = (seconds_delta * 10**9) // 1
+    diff = 0
+    assert nano_delta >= 1, "Must have a period of more than 1 nanosecond."
+
+    if seconds_delta >= 5:  # Save time for long periods by ignoring nano
```

Comment:
Just so I understand the logic: If the desired freq was 1Hz, then the period would be 1 second. According to this, the delta would be 1, however, you would use the get_nano_diff function to compare this. The difference in nano seconds is likely to be quite large but you're comparing this to a delta of 1? This doesn't seem right.

Reply:
Reviewed in person, the delta is set to seconds or nano seconds alongside the diff function that will be used to calculate it